program-runtime/src/invoke_context.rs

```diff
@@ -643,37 +650,48 @@ impl<'a> InvokeContext<'a> {
         program_indices: &[usize],
         account_indices: &[usize],
         caller_write_privileges: &[bool],
-    ) -> Result<u64, InstructionError> {
+    ) -> ProcessInstructionResult {
         let is_lowest_invocation_level = self.invoke_stack.is_empty();
         if !is_lowest_invocation_level {
             // Verify the calling program hasn't misbehaved
-            self.verify_and_update(instruction, account_indices, caller_write_privileges)?;
+            let result =
+                self.verify_and_update(instruction, account_indices, caller_write_privileges);
+            if result.is_err() {
+                return ProcessInstructionResult {
+                    compute_units_consumed: 0,
```

Comment:
Ideally we should also be charging for some of this work, we don't currently but returning zero here is a little strange.  Work for a different PR

Reply:
Added issue: https://github.com/solana-labs/solana/issues/22073 to track this